### PR TITLE
Medication change table

### PR DESCRIPTION
**Pull request by jessica-oflanagan**
This PR addresses Jiras 1206 and 1192 to display medication-changes and the end dates of stopped medications in tabular view.  

When a medication is stopped, its medication-change in the tabular view is red, which required adding functionality to format a particular cell.  I added a property 'formatFunction' to the medication table in VisualizerManager, and TabularListVisualizer uses this function to give a cell a different css class.

**Running the Application**

- [x] Manually tested in Chrome
- Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- Cheat sheet is updated
- Demo script is updated 
- Documentation on Wiki has been updated 
- Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- Added Enzyme-UI tests for slim mode 
- Added Enzyme-UI tests for full mode
- Added backend tests for new functionality added
- Added note parser examples to test new functionality

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by mgramigna** [comment]
I have a clarifying question: if a medication already has an end date in the data by default (like in Debra's case), should the "stopped" red text also be showing up? Or are we only considering the case where the stopping of medication is entered in a note

**Comment by gregquinn2001** [comment]
If a medication just runs its expected course and ends, that is not stopped. Stopped is a change made after the medication was started where the patient will stop taking it before it was originally planned to end.

**Comment by gregquinn2001** [comment]
Ending a medication is because of normal standards of care while stopping it has a reason like toxicities were too much to handle

**Comment by nicoleng12** [comment]
The table looks really nice! I like that there is a new column now for medication change and the end date gets populated when you stop a medication. I noticed a couple of things

[NN1] - Open source note is broken. For example, if I create a new note and do @condition (invasive..) then type #disease status #stable #imaging, close the note, click on #stable in the targeted data panel, select "open source note", there's an error in the console "cannot read property 'entryId' of undefined PatientRecord

[NN2] - Open source note does not work in the medication table. Not sure if this bug is related to NN1 or something different. When I create a new note and enter #stop medication @active medication and choose PAClitaxel, close note, then click on a PACLitaxel in targeted data panel and choose "open source note", a message pops up that no source not available. This should open the note that I just created

File: src/containers/FullApp.jsx [hunk 1]
```diff
@@ -226,7 +226,7 @@ export class FullApp extends Component {
             });
             return;
         }
-        const sourceNote = this.state.patient.getEntryFromReference(item.value[2]);
+        const sourceNote = this.state.patient.getEntryFromReference(item[2]);
```
Comment (mgramigna) [inline src/containers/FullApp.jsx]: NN1 might stem from this change?

**Comment by gregquinn2001** [comment]
@jessica-oflanagan so are we adding a sourceclinicalnote to the MedicationChange object and then referencing that in the end date column if it's stopped or are we referencing the sourceclinicalnote in the MedicationRequested object?

**Comment by jessica-oflanagan** [comment]
@gregquinn2001 We are referencing sourceClinicalNote in the MedicationChange object!

**Comment by nicoleng12** [comment]
NN1 and NN2 are fixed. Looks good! Just needs a rebase and then I will merge
